internal/settings/providers/dynu/provider.go

```diff
@@ -0,0 +1,166 @@
+package dynu
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/qdm12/ddns-updater/internal/models"
+	"github.com/qdm12/ddns-updater/internal/settings/constants"
+	"github.com/qdm12/ddns-updater/internal/settings/errors"
+	"github.com/qdm12/ddns-updater/internal/settings/headers"
+	"github.com/qdm12/ddns-updater/internal/settings/utils"
+	"github.com/qdm12/ddns-updater/pkg/publicip/ipversion"
+)
+
+type provider struct {
+	domain        string
+	host          string
+	group         string
+	ipVersion     ipversion.IPVersion
+	username      string
+	password      string
+	useProviderIP bool
+}
+
+func New(data json.RawMessage, domain, host string,
+	ipVersion ipversion.IPVersion) (p *provider, err error) {
+	extraSettings := struct {
+		Username      string `json:"username"`
+		Password      string `json:"password"`
+		UseProviderIP bool   `json:"provider_ip"`
+		Group         string `json:"group"`
+	}{}
+	if err := json.Unmarshal(data, &extraSettings); err != nil {
+		return nil, err
+	}
+	p = &provider{
```

Comment:
Please default the host to `"@"` and remove the error `ErrEmptyHost`

```suggestion

	if host == "" {
		host = "@" // default
	}
	
	p = &provider{
```